pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -133,18 +133,22 @@ func IsSystemdStyleName(name string) bool {
 type libcontainerAdapter struct {
 	// cgroupManagerType defines how to interface with libcontainer
 	cgroupManagerType libcontainerCgroupManagerType
+	rootless          bool
 }
 
 // newLibcontainerAdapter returns a configured libcontainerAdapter for specified manager.
 // it does any initialization required by that manager to function.
-func newLibcontainerAdapter(cgroupManagerType libcontainerCgroupManagerType) *libcontainerAdapter {
-	return &libcontainerAdapter{cgroupManagerType: cgroupManagerType}
+func newLibcontainerAdapter(cgroupManagerType libcontainerCgroupManagerType, rootless bool) *libcontainerAdapter {
+	return &libcontainerAdapter{cgroupManagerType: cgroupManagerType, rootless: rootless}
 }
 
 // newManager returns an implementation of cgroups.Manager
 func (l *libcontainerAdapter) newManager(cgroups *libcontainerconfigs.Cgroup, paths map[string]string) (libcontainercgroups.Manager, error) {
 	switch l.cgroupManagerType {
 	case libcontainerCgroupfs:
+		if l.rootless {
```

Comment:
do we want to address using cgroupfs and running in a new cgroup namespace?

Reply:
Could you be more specific? I think we are going to support only systemd (as in Podman)

Reply:
I mean we can create a separate environment with cgroup fully owned by the rootless user, where we run both kubernetes and the container runtime.
Similar to:
```
$ systemd-run --user --scope unshare -rCm bash
# mount -t cgroup2 cgroup2 /tmp && mount --move /tmp /sys/fs/cgroup
```
Now /sys/fs/cgroup is fully owned by the rootless user and we can use the cgroupfs backend in the same way as root.

Wouldn't that work?

Reply:
```console
$ cat /etc/systemd/system/user@.service.d/delegate-cgroup.conf 
[Service]
Delegate=pids memory cpu cpuset io
$ cat /sys/fs/cgroup/user.slice/user-1001.slice/user@1001.service/cgroup.controllers 
cpuset cpu io memory pids
$ systemd-run --user --scope unshare -rCm bash
Running scope as unit: run-r31deaab786d54940adb0d0d9e7565247.scope
#  mount -t cgroup2 cgroup2 /tmp && mount --move /tmp /sys/fs/cgroup
mount: /sys/fs/cgroup: filesystem was mounted, but failed to update userspace mount table.
# cat /sys/fs/cgroup/cgroup.controllers 
memory pids
```

Controllers are not delegated as expected

Reply:
@giuseppe 

Can't write to subtree_control
```console
$ systemd-run --user --scope -p "Delegate=cpuset cpu io memory pids" unshare -rCm sh -c "mount -t cgroup2 cgroup2 /tmp && mount -n  --move /tmp /sys/fs/cgroup && cd /sys/fs/cgroup && bash"
Running scope as unit: run-rc35e9a40b0f64c76aa38bc81ce83eb43.scope
# cat cgroup.controllers 
cpuset cpu io memory pids
# cat cgroup.type 
domain
# echo +memory > cgroup.subtree_control 
bash: echo: write error: Device or resource busy
```

Reply:
cgroupns doesn't work expected when containerd and rootlesskit are executed as separate services and `/sys/fs/cgroup` is mounted with `nsdelegate`: https://github.com/rootless-containers/usernetes/pull/178#issuecomment-656039720